Model/Transaction.php

```diff
@@ -113,17 +114,17 @@ public function getChildTransactions()
     }
 
     /**
-     * Returns true if transaction's payment method is Payment on Invoice or Payment in Advance
+     * Returns true if transaction's payment method is "wiretransfer"
      *
      * @return bool
      *
      * @since 1.3.0
      */
     public function isPoiPiaPaymentMethod()
     {
-        $sPaymentId = $this->getPaymentType();
-        $oPayment = PaymentMethodHelper::getPaymentById($sPaymentId);
-        return is_subclass_of($oPayment->getPaymentMethod(), BasePoiPiaPaymentMethod::class);
+        $oXml = simplexml_load_string($this->getResponseXML());
```

Comment:
You have a method getTransactionPaymentMethodName() why not use it?

Reply:
isPoiPiaPaymentMethod() is used only in situations when order reference in transaction object is empty. getTransactionPaymentMethodName() is used in all situations no matter if order reference is empty or not, and it does not target POI/PIA transactions only.

Reply:
here you are creating waste as the same logic is already in the getTransactionPaymentMethodName(). Call the getTransactionPaymentMethodName() and then check the result is the same as the payment method code.

Reply:
Done 👍